src/Symfony/Component/Mailer/EventListener/DkimSignedMessageListener.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Mailer\Event\MessageEvent;
+use Symfony\Component\Mailer\Exception\InvalidArgumentException;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Crypto\DkimSigner;
+use Symfony\Component\Mime\Crypto\SMimeSigner;
+use Symfony\Component\Mime\Message;
+
+/**
+ * Signs the message.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Elías Fernández
+ */
+class DkimSignedMessageListener implements EventSubscriberInterface
+{
+
+    public function __construct(
+        private ?string $key = null,
+        private ?string $domain = null,
+        private ?string $select = null,
+        private ?string $passphrase = null,
+        private array $options = []
+    ) {
+    }
+
+
+    public function onMessage(MessageEvent $event): void
+    {
+        $message = $event->getMessage();
+        if (
+            !$message instanceof Message ||
+            empty($this->key)
+        ) {
+            return;
+        }
+
+        if (empty($this->domain)) {
+            throw new InvalidArgumentException('No domain provided for DKIM message');
+        }
+
+        $signer = new DkimSigner($this->key, $this->domain, $this->select, $this->options, $this->passphrase ?? '');
```

Comment:
it would make more sense to inject a DkimSigner in the listener instead of injecting the needed arguments and instantiating it here.